### DISCUSSION TITLE
Time Series data and cesium primitives

### DISCUSSION THREAD

**Discussion by sleagle** [TerriaJS/terriajs#6492] [Q&A]
Hi,
I am consuming a wfs service which has time series data: 
 "occupy_start": 1853,
 "occupy_end": 1858,

When using TableTimeStyleTraits it works perfectly, however due to different categories I used perPropertyStyles this stuffs up the Time Series data and if use the timeProperty it will only show data at a point will not be able to show the data for the time period. Part of the current configuration is shown below
`"defaultStyle": {
            "time": {
              "timeColumn": "occupy_start",
              "endTimeColumn": "occupy_end",
              "idColumns": ["_id_"],
              "spreadStartTime": false,
              "spreadFinishTime": false
            },
            "hidden": true
          },
          "timeProperty": "occupy_start",
          "perPropertyStyles": [
            {
              "properties": { "type":"nursery"},
              "style": {
                "marker-size": "medium",
                "marker-symbol": "bus"
              },
              "caseSensitive": false
            },
            {
              "properties": { "type":"House of Correction"},
              "style": {
                "marker-size": "medium",
                "marker-symbol": "prison",
                "marker-color": "blue"
              },
              "caseSensitive": false
            },
            {
              "properties": {"type":"road/bridge station"},
              "style": {
                "marker-size": "medium",
                "marker-symbol": "car",
                "marker-color": "green"
              },
              "caseSensitive": false
            },
            {
              "properties": {"type":"bridge station"},
              "style": {
                "marker-size": "medium",
                "marker-symbol": "b",
                "marker-color": "yellow"
              },
              "caseSensitive": false
            }
          ],`

Also would like to know if its possible to achieve clustering as follows: https://sandcastle.cesium.com/gallery/Clustering.html

Any help is highly appreciated

Thank you

**Comment by nf-s** [comment]
Hi @sleagle 

At the moment, you aren't able to mix `TableTimeStyleTraits` with `perPropertyStyles` - you can only use Table styling **or** GeoJson styling.

But, you can use the new `TablePointStyleTraits` to style your markers.

This can be done using new "Edit Style" function.

- Click "Edit Style"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037496-ee76ca2c-d275-4247-be0d-69c181e71e1f.png">
- Set style to your liking
- Click "show advanced options"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037527-eb76afec-517f-4877-bd5a-dc946f982089.png">
- Click "Copy user stratum to clipboard"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037547-5471f26b-2e55-41dd-a9d9-0a3c89835807.png">
- Merge copied JSON into your catalog item JSON

**Note** this feature is brand new and experimental - so you may run into bugs/quirks

As for clustering, at the moment Terria doesn't have any clustering functionality build into the TableStyling or GeoJsonStyling.

It might be possible to cluster features using `czmlTemplate` - but this can't be used in conjuction with TableStyling or GeoJsonStyling.

There is an example of `czmlTemplate` in this discussion question https://github.com/TerriaJS/terriajs/discussions/6118#discussioncomment-2882017

**Reply by sleagle** [reply]
@nf-s 
Thanks for the response will try it out
